lib/dialects/postgres.ts

```diff
@@ -18,32 +18,44 @@ type RawColumn = {
   table_schema: string;
   data_type: string;
   column_default: any | null;
-  character_maximum_length: number | null;
+  character_maximum_length: null | number | string;
   is_generated: 'NEVER' | 'ALWAYS';
   is_nullable: 'YES' | 'NO';
   is_unique: boolean;
   is_primary: boolean;
   is_identity: 'YES' | 'NO';
   generation_expression: null | string;
-  numeric_precision: null | number;
-  numeric_scale: null | number;
+  numeric_precision: null | number | string;
+  numeric_scale: null | number | string;
   serial: null | string;
   column_comment: string | null;
   foreign_key_schema: null | string;
   foreign_key_table: null | string;
   foreign_key_column: null | string;
 };
 
+function convertStringOrNumber(t: string | number | null): number|null {
+  switch (typeof t) {
+    case 'string':
+      return Number(t)
+
+    case 'number':
+      return t
+  }
+
+  return null
+}
+
```

Comment:
```ts
return t == undefined ? t : Number(t)
```
This check should be enough.

Reply:
what about the null value? so in that case it would not be able to return null in the future.

```ts
return t == undefined ||  t === null ? null : Number(t)
```

Reply:
missed that was with 2 equal signs